include/aca_oam_port_manager.h

```diff
@@ -0,0 +1,55 @@
+// Copyright 2019 The Alcor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef ACA_OAM_PORT_MANAGER_H
+#define ACA_OAM_PORT_MANAGER_H
+
+#include <string>
+#include <unordered_map>
+#include <unordered_set>
+#include <mutex>
+#include <atomic>
+
+using namespace std;
+
+namespace aca_oam_port_manager
+{
+class Aca_Oam_Port_Manager {
+  public:
+  Aca_Oam_Port_Manager();
+  ~Aca_Oam_Port_Manager();
+
+  static Aca_Oam_Port_Manager &get_instance();
+
+  void create_entry_unsafe(uint32_t port_number);
```

Comment:
don't think we need to use uint32_t, we can use uint directly (it will likely be 32bit anyway. This will give compile a chance to optimize it.